src/containers/projects/view/project-header.svelte

```diff
@@ -1,32 +1,74 @@
 <script>
-  import { createEventDispatcher } from 'svelte';
-  import Button from 'ui/button.svelte';
-  import TextField from 'ui/text-field.svelte';
+  import { getContext } from 'svelte';
+  import { prefetch, goto } from '@sapper/app';
+  import { Button, H1, Chip } from 'attractions';
+  import { snackbarContextKey } from 'attractions/snackbar';
   import Labeled from 'ui/labeled.svelte';
+  import Notice from '@/components/common/notice.svelte';
   import TagSelector from '@/components/projects/view/tag-selector.svelte';
-  import { formatDateRange } from '@/utils/date-time-format.js';
+  import ReviewVerdict from '@/components/projects/review/review-verdict.svelte';
+  import FinalizingDialog from '@/components/projects/view/finalizing-dialog.svelte';
+  import DangerConfirmDialog from '@/components/projects/view/danger-confirm-dialog.svelte';
   import { API_HOST_BROWSER } from '@/constants/env.js';
   import ProjectStages from '@/constants/backend/project-lifetime-stages.js';
   import ReviewStatuses from '@/constants/backend/project-review-statuses.js';
+  import { formatDateRange } from '@/utils/date-time-format.js';
+  import * as api from '@/utils/api.js';
 
   export let project;
   export let account;
   export let tags;
-  export let review = false;
-
-  $: isModerator = (
-    account && project.moderators.find(moderator => moderator.email === account.email)
-  ) != null;
+  export let moderatorMode = false;
 
-  let reviewComment = null;
+  let finalizeDialogOpen = false;
+  let projectDeletionDialogOpen = false;
+  const reviewView = getContext('review-mode');
 
-  const projectImageURL = (
-    project.image_id == null ?
+  $: projectImageURL = (
+    $project.image_id == null ?
       '/images/create-project/placeholder.svg'
-    : `${API_HOST_BROWSER}/file/${project.image_id}`
+    : `${API_HOST_BROWSER}/file/${$project.image_id}`
   );
 
-  const dispatch = createEventDispatcher();
+  async function saveTags({ detail: tagIDs }) {
+    try {
+      await api.json(api.patch(`/projects/${$project.id}/tags`, {
+        csrfToken: account.csrf_token,
+        data: tagIDs,
+      }));
+      $project.tags = tagIDs;
+      showSnackbar({ props: { text: 'Tags successfully updated' } });
+    } catch (e) {
+      showSnackbar({ props: { text: 'Something went wrong, try reloading the page' } });
+      console.error(e);
+    }
+  }
+
+  async function submitForReview() {
+    try {
+      await api.json(api.patch(
+        `/projects/${$project.id}/request_review`,
+        { csrfToken: account.csrf_token },
+      ));
+      $project.review_status = ReviewStatuses.PENDING;
+    } catch (e) {
+      showSnackbar({ props: { text: 'Something went wrong, try reloading the page' } });
+      console.error(e);
+    }
+  }
+
+  async function deleteProject() {
+    try {
+      prefetch('/projects');
```

Comment:
If you prefetch before you delete the project, wouldn't that cause the page to render with that project there?

Reply:
Yeah, now that I think about it, `prefetch` in most places is not suitable. Will remove.